src/published-data/published-data.controller.ts

```diff
@@ -296,7 +246,7 @@ export class PublishedDataController {
     const createdPublishedData =
       await this.publishedDataService.create(publishedDataDto);
 
-    return this.convertCurrentToObsoleteSchema(createdPublishedData);
+    return createdPublishedData as unknown as PublishedDataObsoleteDto;
```

Comment:
I believe this casting to `unknown` is necessary, due to the missing overlap? That's a pity...